packages/create-rspack/index.js

```diff
@@ -114,6 +114,9 @@ function copyFolder(src, dst) {
 						}
 					}
 				}
+				if (pkg.name) {
+					pkg.name = dst;
```

Comment:
`dst` may contain sub folder, like this:

Can you try `path.dirname(dst)`?

![image](https://github.com/web-infra-dev/rspack/assets/7237365/21b5104e-1ea9-41ce-99af-67f20e2f3aea)

Reply:
> path.dirname(dst)

I think it should work like this: The user creates an empty folder, runs the CLI command in this empty folder, and then enters the desired package name. The selected template code is then copied into the current empty folder. This is also the reason behind the PR I submitted last time: why it is necessary to check if the current directory is empty. By doing it this way, we don't have to worry about whether the user's input contains subfolders; the input is just the package name, which can either include a scope or be standalone. Please evaluate whether this approach is reasonable. If it is, we can make this change.

Reply:
In projects with a monorepo architecture, the names of many packages are often under a scope.

Reply:
My concern is that this will require users to create an empty directory before running the CLI, which most users will not do.

Reply:
Users have already developed habits, so what we can do is try to adapt to more usage scenarios and align with other tools in the community

Reply:
> Users have already developed habits, so what we can do is try to adapt to more usage scenarios and align with other tools in the community

After using path.dirname(dst), for example, if the user inputs test/rspack-project, the name in the created template's package.json will be test instead of test/rspack-project. In my scenario, I want to use rspack to directly create a project's package.json name in a monorepo project to be exactly what I input. Since my input does not include subfolder names, I think it would be better to use dst directly.

Reply:
I think we can support both of them:

- If input is `subdir/rspack-project`, the package.json#name should be `rspack-project`, the output path is `/cwd/subdir/rspack-project'.
- If input is `@scope/rspack-project,`, the package.json#name should be `@scope/rspack-project`, the output path is `/cwd/rspack-project'.

Reply:
> I think we can support both of them:我认为我们可以支持他们两个：
> 
> * If input is `subdir/rspack-project`, the package.json#name should be `rspack-project`, the output path is `/cwd/subdir/rspack-project'.如果输入是 `subdir/rspack-project` ，则 package.json#name 应为 `rspack-project` ，输出路径为 `/cwd/subdir/rspack-project'。
> * If input is `@scope/rspack-project,`, the package.json#name should be `@scope/rspack-project`, the output path is `/cwd/rspack-project'.如果输入为 `@scope/rspack-project,` ，则 package.json#name 应为 `@scope/rspack-project` ，输出路径为 `/cwd/rspack-project'。

done